app/src/models/repository.ts

```diff
@@ -88,12 +90,37 @@ export type RepositoryWithGitHubRepository = Repository & {
   readonly gitHubRepository: GitHubRepository
 }
 
+export type RepositoryWithForkedGitHubRepository = Repository & {
+  readonly gitHubRepository: ForkedGitHubRepository
+}
+
+/**
+ * Returns whether the passed repository is a GitHub repository.
+ *
+ * This function narrows down the type of the passed repository to
+ * RepositoryWithGitHubRepository if it returns true.
+ */
 export function isRepositoryWithGitHubRepository(
   repository: Repository
 ): repository is RepositoryWithGitHubRepository {
   return repository.gitHubRepository instanceof GitHubRepository
 }
 
+/**
+ * Returns whether the passed repository is a GitHub fork.
+ *
+ * This function narrows down the type of the passed repository to
+ * RepositoryWithForkedGitHubRepository if it returns true.
+ */
+export function isRepositoryWithForkedGitHubRepository(
+  repository: Repository
+): repository is RepositoryWithForkedGitHubRepository {
+  return (
+    isRepositoryWithGitHubRepository(repository) &&
+    repository.gitHubRepository.parent !== null
+  )
+}
```

Comment:
This helper method allows us to be able to pass `RepositoryWithForkedGitHubRepository ` objects around and not have to check whether `parent` is `null` when we now it isn't.